src/integration/java/io/pinecone/integration/dataPlane/pod/QueryErrorTest.java

```diff
@@ -42,26 +42,27 @@ public static void cleanUp() {
 
     @Test
     public void queryWithIncorrectVectorDimensionSync() {
-        int numOfVectors = 3;
-
         String namespace = RandomStringBuilder.build("ns", 8);
-        PineconeBlockingDataPlaneClient dataPlaneClient = new PineconeBlockingDataPlaneClient(blockingStub);
+        Index dataPlaneClient = new Index(blockingStub);
         DescribeIndexStatsResponse describeIndexStatsResponse1 = dataPlaneClient.describeIndexStats(null);
         assertEquals(describeIndexStatsResponse1.getDimension(), dimension);
 
+        StringBuilder exceptionMessage = new StringBuilder();
         // Query with incorrect dimensions
         try {
             List<Float> vector = Arrays.asList(100F);
             dataPlaneClient.query(5, vector, null, null, null, namespace, null, true, true);
         } catch (StatusRuntimeException statusRuntimeException) {
-            assert (statusRuntimeException.getTrailers().toString().contains("grpc-status=3"));
-            assert (statusRuntimeException.getTrailers().toString().contains("grpc-message=Query vector dimension 1 does not match the dimension of the index 3"));
+            exceptionMessage.append(statusRuntimeException.getTrailers().toString());
+        } finally {
```

Comment:
If the exception thrown is not StatusRuntimeException, then the assertions in catch phrase will never execute. In the upcoming PR for adding tests for serverless index, I'll update this for rest of the tests.